### PR TITLE
tests(helpers): separate customized asserts

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

KAG-5252

- move assertions to a separated lua module (only copy, no change)
- move `lookup()` to misc module

### Checklist

- [ ] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by Water-Melon — APPROVED** [review]

File: spec/details/asserts.lua [hunk 1]
```diff
@@ -0,0 +1,823 @@
+local cjson = require("cjson.safe")
```
Comment (fffonion) [inline spec/details/asserts.lua:1]: the naming of `spec/details` feels bit confusing. we might want to call something like `spec/tools` or `spec/internal`. not blocker for this specific PR.

Comment (chronolaw) [reply spec/details/asserts.lua:1]: We will do it in the future, thanks.

**Comment by Tieske** [comment]
please validate that the documentation still renders properly when changing those helper files.

```
cd spec
ldoc .
```

should render the docs properly. Users access this through `pongo docs`.

**Comment by chronolaw** [comment]
> please validate that the documentation still renders properly when changing those helper files.
> 
> ```
> cd spec
> ldoc .
> ```
> 
> should render the docs properly. Users access this through `pongo docs`.

See comment https://github.com/Kong/kong/pull/13630#issuecomment-2333872831

File: spec/details/asserts.lua (hunk 1 above)
Comment (kikito) [reply spec/details/asserts.lua:9]: I agree that `details` is not a good name.

Comment (chronolaw) [reply spec/details/asserts.lua:9]: I will rename this directory after these PRs are merged.

**Review by ProBrian — APPROVED** [review]

**Pull request merged**